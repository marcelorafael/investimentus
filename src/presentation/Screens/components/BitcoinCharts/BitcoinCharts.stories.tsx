import React from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { View, Text } from 'react-native';
import { BitcoinCharts } from './BitcoinCharts'

storiesOf('Screens/BitcoinCharts', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <BitcoinCharts title={text('title','Crypto Moedas')}>
      <View>
        <Text style={{backgroundColor:"#fff", borderRadius: 5}}>fbgnhmj,hmgfdsasdfgh</Text>
        <Text style={{backgroundColor:"#fff", borderRadius: 5}}>fbgnhmj,hmgfdsasdfgh</Text>
        <Text style={{backgroundColor:"#fff", borderRadius: 5}}>fbgnhmj,hmgfdsasdfgh</Text>
        <Text style={{backgroundColor:"#fff", borderRadius: 5}}>fbgnhmj,hmgfdsasdfgh</Text>
      </View>
    </BitcoinCharts>
  ))
