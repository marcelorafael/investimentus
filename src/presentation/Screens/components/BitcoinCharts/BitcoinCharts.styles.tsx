import styled from 'styled-components/native'
import LinearGradient from 'react-native-linear-gradient'

export const Container = styled(LinearGradient).attrs({
  colors: ['#000', '#7F5DF0']
})`
  flex: 1;

  padding-left: 12px;
  padding-right: 12px;
`

export const Header = styled.View`
  flex-direction: row;
  justify-content: space-between;

  padding: 8px;
`

export const Title = styled.Text`
  ${({ theme }) => theme?.FONTS?.bullpen ?? {}};
  color: ${({ theme }) => theme?.COLORS?.white ?? '#FFF'};
`
