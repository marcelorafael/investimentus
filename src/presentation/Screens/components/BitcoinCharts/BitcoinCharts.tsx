import React from 'react'
import Icon from 'react-native-vector-icons/FontAwesome'
import { Container, Header, Title } from './BitcoinCharts.styles'

interface BitcoinChartsProps {
  title: string
  children: React.ReactElement
}

const BitcoinCharts: React.FC<BitcoinChartsProps> = (
  props: BitcoinChartsProps
) => {
  const { children, title } = props
  return (
    <Container>
      <Header>
        <Title>{title}</Title>
        <Icon name="bitcoin" color="#FFF" size={32} />
      </Header>
      {children}
    </Container>
  )
}
const BitcoinChartsMemo = React.memo(BitcoinCharts)
export { BitcoinChartsMemo as BitcoinCharts }
