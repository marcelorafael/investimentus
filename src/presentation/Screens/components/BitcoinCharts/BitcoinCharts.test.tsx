import React from 'react'
import renderer from 'react-test-renderer'
import { BitcoinCharts } from './BitcoinCharts'

describe('Screens/BitcoinCharts', () => {
  it('should match snapshot', () => {
    const sut = (
      <BitcoinCharts title="">
        <></>
      </BitcoinCharts>
    )

    const renderComponent = renderer.create(sut)

    expect(renderComponent).toMatchSnapshot()
  })
})
