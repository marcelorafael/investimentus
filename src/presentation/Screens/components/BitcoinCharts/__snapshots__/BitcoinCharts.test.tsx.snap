// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens/BitcoinCharts should match snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278190080,
      4286537200,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingLeft": 12,
        "paddingRight": 12,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
          },
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 32,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï…š
    </Text>
  </View>
</BVLinearGradient>
`
